src/osversion/VersionTranslator.cpp

```diff
@@ -7664,6 +7665,239 @@ namespace osversion {
+          "SupplyAirFlowRate",      "FlowPerFloorArea",      "FractionOfAutosizedCoolingValue", "FractionOfAutosizedHeatingValue",
+          "FlowPerCoolingCapacity", "FlowPerHeatingCapacity"};
+
+        constexpr std::array<std::string_view, 6> noCoolHeatSAFMethodChoicesUC{
+          "SUPPLYAIRFLOWRATE",      "FLOWPERFLOORAREA",      "FRACTIONOFAUTOSIZEDCOOLINGVALUE", "FRACTIONOFAUTOSIZEDHEATINGVALUE",
+          "FLOWPERCOOLINGCAPACITY", "FLOWPERHEATINGCAPACITY"};
+
+        // Reset all flow fields, we want only one at best to be set
+        for (size_t i = 0; i < coolingSAFMethodChoices.size(); ++i) {
+          const size_t fieldIndex = coolingSAFMethodIndex + 1 + i;
+          newObject.setString(fieldIndex, "");
+        }
+        for (size_t i = 0; i < heatingSAFMethodChoices.size(); ++i) {
+          const size_t fieldIndex = heatingSAFMethodIndex + 1 + i;
+          newObject.setString(fieldIndex, "");
+        }
+        for (size_t i = 0; i < noCoolHeatSAFMethodChoices.size(); ++i) {
+          const size_t fieldIndex = noCoolHeatSAFMethodIndex + 1 + i;
+          newObject.setString(fieldIndex, "");
+        }
+
+        if (!hasCoolingCoil) {
+          newObject.setString(coolingSAFMethodIndex, "None");
+        } else if (!coolingSAFMethod) {
+          // Technically here E+ checks if the cooling coil rated air flow rate is autosized or not
+          // Also could get overriden by the DesignSpecification:ZoneHVAC:Sizing
+          // Finally it lets it slide as long as one of the other SupplyAirFlowRateXXX is not empty
+          // Here I guess we'll scan the fields in order, and pick the first we find
+          // * 19 - Supply Air Flow Rate During Cooling Operation
+          // * 20 - Supply Air Flow Rate Per Floor Area During Cooling Operation
+          // * 21 - Fraction of Autosized Design Cooling Supply Air Flow Rate
+          // * 22 - Design Supply Air Flow Rate Per Unit of Capacity During Cooling Operation
+          bool found = false;
+          for (size_t i = 0; i < coolingSAFMethodChoices.size(); ++i) {
+            const size_t fieldIndex = coolingSAFMethodIndex + 1 + i;
+            if ((value = object.getString(fieldIndex))) {
+              newObject.setString(coolingSAFMethodIndex, std::string{coolingSAFMethodChoices[i]});
+              newObject.setString(fieldIndex, *value);
+              found = true;
+              break;
+            }
+          }
+          if (!found) {
+            LOG(Warn, "For AirLoopHVACUnitarySystem '"
+                        << object.nameString()
+                        << "', there is a cooling coil, but no Supply Air Flow Rate Method During Cooling Operation and no flow rate fields set. "
+                           "Defaulting to SupplyAirFlowRate with Autosize");
+            newObject.setString(coolingSAFMethodIndex, "SupplyAirFlowRate");
+            newObject.setString(coolingSAFMethodIndex + 1, "Autosize");
+          }
```

Comment:
If you have a cooling coil, but no Cooling SAF method, first the first flow field that isn't empty, set the method to match, and copy that flow field.

Example

In:

```
  ,                                       !- Supply Air Flow Rate Method During Cooling Operation
  ,                                       !- Supply Air Flow Rate During Cooling Operation {m3/s}
  1.0,                                  !- Supply Air Flow Rate Per Floor Area During Cooling Operation {m3/s-m2}
  3.0,                                  !- Fraction of Autosized Design Cooling Supply Air Flow Rate
  ,                                       !- Design Supply Air Flow Rate Per Unit of Capacity During Cooling Operation {m3/s-W}

```

=> out

```
  FlowPerFloorArea,       !- Supply Air Flow Rate Method During Cooling Operation
  ,                                       !- Supply Air Flow Rate During Cooling Operation {m3/s}
  1.0,                                  !- Supply Air Flow Rate Per Floor Area During Cooling Operation {m3/s-m2}
  ,                                       !- Fraction of Autosized Design Cooling Supply Air Flow Rate
  ,                                       !- Design Supply Air Flow Rate Per Unit of Capacity During Cooling Operation {m3/s-W}
```

Reply:
If you can't find any, just default to "SupplyAirFlowRate" and "Autosize"